core/src/main/java/com/linecorp/armeria/client/ClientOptions.java

```diff
@@ -208,6 +209,30 @@ private ClientOptions(ClientOptions baseOptions, ClientOptions additionalOptions
         super(baseOptions, additionalOptions);
     }
 
+    @Override
+    protected <T extends AbstractOptionValue<?, ?>> T mergeValue(T oldValue, T newValue) {
+        if (oldValue.option() == DECORATION) {
+            final ClientDecoration oldDecoration = (ClientDecoration) oldValue.value();
+            final ClientDecoration newDecoration = (ClientDecoration) newValue.value();
```

Comment:
Could return oldDecoration or newDecoration if newDecoration or oldDecoration is empty respectively.